bvm/ballerina-core/src/main/java/org/ballerinalang/connector/impl/ResourceExecutor.java

```diff
@@ -130,9 +135,10 @@ public static void execute(Resource resource, BServerConnectorFuture connectorFu
                     doubleRegs[doubleParamCount++] = ((BFloat) value).floatValue();
                 } else if (btype == BTypes.typeInt) {
                     longRegs[longParamCount++] = ((BInteger) value).intValue();
-                } else if (value instanceof BStruct) {
-                    refRegs[refParamCount++] = (BRefType) value;
-                } else if (value instanceof BRefValueArray) {
+                } else if (btype == BTypes.typeBlob) {
+                    byteRegs[byteParamCount++] = ((BBlob) value).blobValue();
+                } else if (value instanceof BStruct || value instanceof BRefValueArray || value instanceof BJSON ||
+                        value instanceof BXML) {
```

Comment:
Shall we use BTypes.typeJSON and typeJSON.typeXML here?

Reply:
Addressed with b98823c.